angular-workspace/projects/ni/nimble-angular/src/directives/anchor-menu-item/nimble-anchor-menu-item-router-link-with-href.directive.ts

```diff
@@ -0,0 +1,23 @@
+import { Directive, Input } from '@angular/core';
+import { RouterLinkWithHref } from '@angular/router';
+
+/**
+ * Selectors used for built-in Angular RouterLink directives:
+ * RouterLink: ':not(a):not(area)[routerLink]'
+ * RouterLinkWithHref: 'a[routerLink],area[routerLink]'
+ *
+ * See https://github.com/angular/angular/blob/5957ff4163f55d814be2cf80b9909244f1ce5262/packages/router/src/directives/router_link.ts
+ */
+
+/**
+ * Directive to handle nimble-anchor-menu-item RouterLink support.
+ * Note: Clients need to use [nimbleRouterLink] instead of [routerLink], so that there
+ * won't also be an active RouterLink directive incorrectly handling navigation.
+ */
+@Directive({ selector: 'nimble-anchor-menu-item[nimbleRouterLink]' })
+export class NimbleAnchorMenuItemRouterLinkWithHrefDirective extends RouterLinkWithHref {
+    @Input()
+    public set nimbleRouterLink(commands: never[] | string | null | undefined) {
```

Comment:
Why are we using `never` here instead of `unknown`?

Reply:
Honestly, I don't know. It was what we did in the breadcrumb item's version of this directive, and I've been copying it ever since. I don't even have a sense of how `never[]` is different that `unknown[]` or `any[]`. For consistency, I'd lean towards leaving it this way, but I'm happy to change them all if there's a reason to.

Reply:
I think you can probably leave it like this, for now, but I would bet that this (and everywhere else we're using `never`) should eventually be changed to `unknown`. One interesting question is have we even tested the scenario where `commands` is provided as an array, or if we've only tested the case where it's a string.

Reply:
I think you can probably leave it like this, for now, but I would bet that this (and everywhere else we're using `never`) should eventually be changed to `unknown`. One interesting question is have we even tested the scenario where `commands` is provided as an array, or if we've only tested the case where it's a string.

Reply:
That's a good point. I'm not sure I understand that use case in the first place.